src/bpmat/KSM.cpp

```diff
@@ -928,6 +932,9 @@ int GMRES::solve(TACSVec *b, TACSVec *x, int zero_guess) {
       x->axpy(1.0, W[0]);
     }
 
+    iterCount += niters;
+    resNorm = res[niters];
```

Comment:
You should take the absolute value of this. "res" really stores the solution here since the Givens rotations are applied to it. The norm should always be positive.

Reply:
Should I move these lines above this solution computation block to get the true residual then?